frontend/packages/knative-plugin/src/components/functions/QuickStartGettingStartedCard.tsx

```diff
@@ -0,0 +1,129 @@
+  getQuickStartStatus,
+} from '@patternfly/quickstarts';
+import { RouteIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import QuickStartsLoader from '@console/app/src/components/quick-starts/loader/QuickStartsLoader';
+import {
+  GettingStartedCard,
+  GettingStartedLink,
+} from '@console/shared/src/components/getting-started';
+
+interface QuickStartGettingStartedCardProps {
+  featured?: string[];
+  title?: string;
+  description?: string;
+  filter?: (QuickStart) => boolean;
+}
+
+const orderQuickStarts = (
+  allQuickStarts: QuickStart[],
+  allQuickStartStates: AllQuickStartStates,
+  featured: string[],
+  filter?: (QuickStart) => boolean,
+): QuickStart[] => {
+  const orderedQuickStarts: QuickStart[] = [];
+
+  const filteredQuickStarts = filter ? allQuickStarts.filter(filter) : allQuickStarts;
+
+  const getStatus = (quickStart: QuickStart) =>
+    getQuickStartStatus(allQuickStartStates, quickStart.metadata.name);
+
+  // Prioritize featured quick starts and keep specified order
+  if (featured) {
+    const featuredQuickStartsByName = filteredQuickStarts.reduce((acc, q) => {
+      acc[q.metadata.name] = q;
+      return acc;
+    }, {} as Record<string, QuickStart>);
+    featured.forEach((quickStartName) => {
+      if (
+        featuredQuickStartsByName[quickStartName] &&
+        getStatus(featuredQuickStartsByName[quickStartName]) !== QuickStartStatus.COMPLETE
+      ) {
+        orderedQuickStarts.push(featuredQuickStartsByName[quickStartName]);
+      }
+    });
+  }
+
+  return orderedQuickStarts;
+};
+
+export const QuickStartGettingStartedCard: React.FC<QuickStartGettingStartedCardProps> = ({
+  featured,
+  title,
+  description,
+  filter,
+}) => {
+  const { t } = useTranslation();
+  const { allQuickStartStates, setActiveQuickStart } = React.useContext<QuickStartContextValues>(
+    QuickStartContext,
+  );
+
+  return (
+    <QuickStartsLoader>
+      {(quickStarts, loaded) => {
+        const orderedQuickStarts = orderQuickStarts(
+          quickStarts,
+          allQuickStartStates,
+          featured,
+          filter,
+        );
+        const slicedQuickStarts = orderedQuickStarts.slice(0, 2);
+
+        let links: GettingStartedLink[] = [];
+        if (loaded && slicedQuickStarts.length === 0) {
+          links.push(
+            {
+              id: 'ide-extensions',
+              href:
+                'https://marketplace.visualstudio.com/items?itemName=redhat.vscode-openshift-connector',
+              title: t('knative-plugin~Create using IDE extension'),
+              external: true,
+            },
+            {
+              id: 'functions-tekton-pipelines',
+              title: t('knative-plugin~Building Functions on Cluster with Tekton Pipelines'),
+              href:
+                'https://github.com/knative/func/blob/main/docs/building-functions/on_cluster_build.md',
+              external: true,
+            },
+          );
```

Comment:
Can we add one link if we found just QS, and both if there are 0?

Reply:
It is discussed as either QS will be there or external links. One QS is there also, external link will be added to the 3rd section, which is documentation section.